src/core_landice/mode_forward/mpas_li_thermal.F

```diff
@@ -2037,7 +2115,7 @@ subroutine li_temperature_to_enthalpy(&
 
       do k = 1, nVertLevels
          enthalpy(k) = (1.0_RKIND - waterfrac(k)) * rhoi * cp_ice * temperature(k)   &
-                      + waterfrac(k) * rho_water * (cp_ice * pmpTemperature(k) + latent_heat_ice)
+                      + waterfrac(k) * rhoi * (cp_ice * pmpTemperature(k) + latent_heat_ice)
```

Comment:
We should use rhoi. Here is the reason: from Eqn (6) in the Aschwanden et al. (2012) paper "An enthalpy formulation for glaciers and ice sheets", 

rho * H = rho_i * H_i + rho_w * H_w, 

where rho_i and rho_w are not the ice and water density, but the partial density, i.e., rho_i = mass_ice/volume, rho_w = mass_water/volume. Because water fraction is very small in general, rho_i approximates to rhoi (ice density), rho_w is close to 0. 
Also note that in MALI the enthalpy is equal to the enthalpy density (rho*H) in Aschwanden's paper (they don't have density in their enthaly definition, but we do). Thus, Let's denote MALI's enthalpy as E, and denote W as the water content, then

E = rho * H = rho_i * H_i + rho_w * H_w = (1-W) * H_i + W * H_w
E = (1-W) * rho * cp * T + W * (rho*cp*T_pmp + rho*L)

As rho approximates to rhoi, we have

E = (1-W) * rhoi * cp * T + W * rhoi * (cp*T_pmp + L)

This is what we get in the code.

Reply:
Ok, I think I follow that.  How did you determine that this was incorrect?

Reply:
I can't remember how...I guess it's because Bill's code is so nice that I had to focus on the trivial things and dig something out to prove that I was indeed working on the code...

Reply:
@tongzhangice , I went through the math myself, following the Aschwanden et al. paper and agree that this should be rhoi as it approximates the density of the mixture.  I think you should add a comment here mentioning that and referring to Eq. 1-11 in Aschwanden to explain it.

Reply:
Comments are added in the code